spec-tests/src/test/java/org/candlepin/spec/ExportSpecTest.java

```diff
@@ -0,0 +1,649 @@
+        private ZipFile export;
+        private File manifest;
+        private Map<String, ProductDTO> productIdToProduct = new HashMap<>();
+
+        @BeforeAll
+        void beforeAll() throws Exception {
+            client = ApiClients.admin();
+            ownerApi = client.owners();
+            ownerContentApi = client.ownerContent();
+            ownerProductApi = client.ownerProducts();
+            cdnApi = client.cdns();
+            consumerApi = client.consumers();
+            consumerTypeApi = client.consumerTypes();
+            rulesApi = client.rules();
+            rolesApi = client.roles();
+            usersApi = client.users();
+            jobsApi = client.jobs();
+
+            initializeData();
+
+            manifest = createExport(client, consumer.getUuid(), cdn);
+            export = ExportUtil.getExportArchive(manifest);
+        }
+
+        @AfterAll
+        void afterAll() throws Exception {
+            export.close();
+        }
+
+        @Test
+        @DisplayName("should export consumer types")
+        void shouldExportConsumerTypes() throws Exception {
+            List<ConsumerTypeDTO> expectedConsumerTypes = consumerTypeApi.getConsumerTypes();
+            for (ConsumerTypeDTO expected : expectedConsumerTypes) {
+                String path = CONSUMER_TYPE_PATH + expected.getLabel() + ".json";
+                ConsumerTypeDTO actual = ExportUtil.deserializeJsonFile(export, path, ConsumerTypeDTO.class);
+                assertNotNull(actual);
+                assertEquals(expected.getId(), actual.getId());
+                assertEquals(expected.getLabel(), actual.getLabel());
+            }
+        }
+
+        @Test
+        @DisplayName("should export consumers")
+        void shouldExportConsumers() throws Exception {
+            String path = EXPORT_PATH + "consumer.json";
+            ConsumerDTO actual = ExportUtil
+                .deserializeJsonFile(export, path, ConsumerDTO.class);
+            assertNotNull(actual);
+            assertEquals(consumer.getUuid(), actual.getUuid());
+            assertEquals(consumer.getName(), actual.getName());
+        }
+
+        @Test
+        @DisplayName("should exports CDN URL")
+        void shouldExportCdnUrl() throws Exception {
+            ZipEntry metaFile = export.getEntry(EXPORT_PATH + "meta.json");
+            assertNotNull(metaFile);
+            try (InputStream istream = export.getInputStream(metaFile)) {
+                JsonNode root = ApiClient.MAPPER.readTree(istream);
+                String actualCdnLabel = root.get("cdnLabel").asText();
+                assertEquals(cdn.getLabel(), actualCdnLabel);
+            }
+        }
+
+        @Test
+        @DisplayName("should not include consumer json in entitlements")
+        void shouldNotIncludeConsumerJsonInEntitlements() throws Exception {
+            int target = ENTITILEMENTS_PATH.length() - 1;
+            AtomicInteger count = new AtomicInteger(0);
```

Comment:
Is this a workaround for parallel streams, or did the compiler complain about it a non-final variable being referenced from the closure?

If it's the former, we could avoid it with a call to `.sequential()` before the filters (though I'm pretty sure `.stream()` on a collection is sequential already).

If it's the latter, a cleaner way to do this would be to drop `.forEach` in exchange for the non-terminal `.peek`:

```
int count = export.stream()
  .sequential() // probably optional/extraneous
  .filter(...)
  .filter(...)
  .peek(entry -> { ... })
  .count();

assertEquals(6, count);
```

This can be applied to all of the methods using this.

Reply:
Updated